app/src/main/java/com/github/orkest/domain/FireStoreDatabaseAPI.kt

```diff
@@ -17,12 +19,18 @@ import com.google.firebase.ktx.Firebase
 import java.time.LocalDateTime
 import java.util.concurrent.CompletableFuture
 
-class FireStoreDatabaseAPI {
+open class FireStoreDatabaseAPI {
 
     companion object{
         val db = Firebase.firestore
     }
 
+    fun isOnline(context: Context): Boolean {
+        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val network = connectivityManager.activeNetwork ?: return false
+        val networkCapabilities = connectivityManager.getNetworkCapabilities(network) ?: return false
+        return networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+    }
```

Comment:
Don't forget to add comments, for example : 

/**
 *checks if the device has an active internet connection
 *@param context The context to access the ConnectivityManager
 *@return true if the device has an active internet connection, false otherwise
 */
 
 
 
 Also, activeNetwork() and getNetworkCapabilities() methods are not available on older Android versions. Maybe do an : 
 
  if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) ... else (deprecated version)

Reply:
Right but if we go for this, we need to apply the same principle application wide. Let's discuss it next meeting. 
I'll add the comments.